### PR TITLE
finagle-redis: Implement SCAN commands for sets and sorted sets

### PR DESCRIPTION
**Pull request by jpgneves**
Problem

finagle-redis does not support the SSCAN and ZSCAN commands for
sets and sorted sets, respectively.

Solution

Implement SSCAN and ZSCAN in finagle-redis.

### CODE REVIEW

File: finagle-redis/src/test/scala/com/twitter/finagle/redis/commands/sorted-set/SortedSetClientIntegrationSuite.scala [hunk 1]
```diff
@@ -12,6 +15,8 @@ import org.scalatest.junit.JUnitRunner
 @RunWith(classOf[JUnitRunner])
 final class SortedSetClientIntegrationSuite extends RedisClientTest {
 
+  val TIMEOUT = 5.seconds
```
Comment (bryce-anderson) [inline finagle-redis/src/test/scala/com/twitter/finagle/redis/commands/sorted-set/SortedSetClientIntegrationSuite.scala]: I really like that you've added a timeout: it seems it has been omitted from the other test cases in the suite.

**Comment by bryce-anderson** [comment]
@jpgneves, this has obviously sat in queue way too long without reply, and for that, I'm really sorry. It looks like good work, which makes it even worse. 😞 

There has been some activity in finagle-redis recently, mot notably commit 792498babeb42. I think it's going to require a few modifications to this PR to get it all ready to merge, if you're still interested.

Again, I'm really sorry that this PR slid by for so long with radio silence.

**Comment by jpgneves** [comment]
@bryce-anderson thanks for the feedback! I will attempt to rebase this branch on master and get it all cleaned up for a second go Soon(tm). :)

**Comment by jpgneves** [comment]
Ah, had not realized 792498b is still in the pipeline. I will attempt to get the branch ready in advance for those changes. :)

**Comment by bryce-anderson** [comment]
@jpgneves, I'm pulling @vkostyukov into the conversation. Vladimir is doing a fair amount of work in this area getting it in better shape so it might be good to coordinate to minimize the number of necessary `git merge` calls. 😉

**Comment by vkostyukov** [comment]
Thanks @bryce-anderson!

Sorry for the silence on this, @jpgneves!

We're in the middle of massive refactoring of finagle-redis as part of our bigger effort to bring N4 support in here. The encoding part is done and merged internally (I think, will show up here on Github quite soon). The decoding part is the one requires more attention.

I'm going to work on decoding this week. I'd recommend not rebasing anything yet, but rather wait until we finish all the dirty work.

I will keep this thread updated with the progress.

**Comment by jpgneves** [comment]
I saw that 6.40 was released with some big-ish changes and I tweaked my branch to reflect those.

Anything else I should wait for, or are we good to take another look at this?

File: finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Sets.scala [hunk 2]
```diff
@@ -41,3 +44,23 @@ case class SRandMember(key: Buf, count: Option[Int] = None) extends StrictKeyCom
...
 }
+
+case class SScan(
+  key: Buf,
+  cursor: Long,
+  count: Option[JLong] = None,
+  pattern: Option[Buf] = None
+  ) extends Command {
+  def name: Buf  = Command.SSCAN
+  def toBuf: Seq[Buf] = {
```
Comment (vkostyukov) [inline finagle-redis/src/main/scala/com/twitter/finagle/redis/protocol/commands/Sets.scala]: Can you rebase your branch. I think it's called `body` in master.

**Comment by vkostyukov** [comment]
Thank you @jpgneves! I think is ready to go alive - we're not planning to touch any encoding-related parts in finagle-redis. Once you rebase you branch against current master, we'll pull that internally.

**Comment by jpgneves** [comment]
Done! :)

**Comment by mosesn** [comment]
@jpgneves sorry for the delay, LGTM!  I'm going to see if I can pull this in.

**Comment by mosesn** [comment]
@jpgneves I'm renaming PATTERN to MATCH per https://github.com/twitter/finagle/pull/587/files

**Comment by jpgneves** [comment]
Hi! Sounds good! Thanks for taking a look at this again! :)

**Comment by mosesn** [comment]
@jpgneves this merged in here: https://github.com/twitter/finagle/commit/1b373df466407605c0c145def10a1ddc2738d002 thanks for the contribution!

**Pull request closed without merging**